### DISCUSSION TITLE
Pull Perl docs into Idea!

### DISCUSSION THREAD

**Discussion by that-guy-scott** [Camelcade/Perl5-IDEA#2756] [Ideas]
I love being able to view API documentation right inside the editor.
What I end up doing is this...
Say I have a CPAN module Data::Dumper. I want to view documentation about this module. Normally one would just google "Data::Dumper". Do this enough times and you memorize the urls. 

https://perldoc.perl.org/Data::Dumper. 
https://perldoc.perl.org/some-perl-module-here

I've had perl.org redirect you to the cpan page for these docs as well. 

Would it be possible to pull this info in much like java docs? It's default shortcut is Ctrl+q

**Comment by rwp0** [comment]
Do you mean online rather than the existing offline documentation support?

As currently it requires the module to be installed locally.

(Offline documentation support already exists in the IDE/plugin, there's a tool window for that.)

**Comment by hurricup** [comment]
@that-guy-scott it should work already. But you must have perl or module installed and configured.

**Comment by that-guy-scott** [comment]
I'm using WSL. I get a popup windows but it says "No Documentation Found". I've installed the plugin in Intellij. I'm not sure how to install it locally. I followed the "Getting Started" in the wiki here ==> https://github.com/Camelcade/Perl5-IDEA/wiki/Getting-started

**Reply by hurricup** [reply]
Hm. this might be the problem. I'll check if it works with remote case, thank you!

**Reply by that-guy-scott** [reply]
I should also clarify that my project is local on windows but I use perl on wsl. I don't do the remote development but load the project local in windows and point the perl interpreter to WSL. I'm also having issues with the autocompletion. I'll get it on 1 project but not on another. I have no idea why. I specify the lib folder in intellij perl settings. I'm still pretty new to perl so I apologize if I'm doing something dumb. lol

**Reply by that-guy-scott** [reply]
So I loaded up my linux box and created a basic project there. I'm able to see the docs now. My guess is WSL.

**Reply by hurricup** [reply]
You may also install strawberry perl on windows. The problem not exactly wsl, but non-local interpreter (wsl/docker)

**Reply by that-guy-scott** [reply]
Thanks guys! I thought I had a great idea but you guys already done it. This plugin is fantastic by the way. I use it everyday. So thank you  guys for all your hard work. I really appreciate it. 
update: Strawberry perl loads the docs for me and fixed all my little issues. WSL isn't quite there yet.

**Reply by hurricup** [reply]
what perl are you using in wsl? is it system perl or plenv/perbrew? 
I can confirm that it is working for perlbrew in docker, for example, but does not work for system perl in docker.
IDE treats docker/wsl as kind of 'remote interpreter'

**Reply by hurricup** [reply]
oh. update:

for some reason `perl:5.38.0` has two perls inside it. Like native system 5.36.0 in `/usr/bin` and additional 5.38.0 in `/usr/local/bin`

The first one seems come without documentation, because searching shows only one doc file. So, probably your perl comes without docs as well. 

However, documentation for non-bundled, additional modules should always work if it is there and works.
